axonserver/src/main/java/io/axoniq/axonserver/access/user/UserController.java

```diff
@@ -61,36 +94,43 @@ private User syncUser(String username, String password, Set<UserRole> roles) {
     }
 
     /**
-     * Updates/creates a user with specified password and roles. If password is set, it will be hashed. If no password
-     * is set and
-     * user already exists the stored password is not changed.
+     * Updates/creates a {@link User} in the database. If a password is set, it will be hashed. If no password
+     * is set and the user already exists the stored password is not changed. If the user does not exist, and the
+     * password is not set, the user will get a {@link #PWD_NOLOGON} marker instead.
      *
-     * @param username the username
-     * @param password the plaintext password
-     * @param roles    the roles granted to the user
-     * @return the stored information, including the hashed password
+     * Note that the {@link #PWD_NOLOGON} marker prevents a match with any password, as it is not hashed.
+     *
+     * @param jpaUser the {@link User} object to store.
      */
-    @Transactional
-    public User updateUser(String username, String password, Set<UserRole> roles) {
-        return syncUser(username, password == null ? null: passwordEncoder.encode(password), roles);
-    }
-
     @Transactional
     public void syncUser(User jpaUser) {
         synchronized (userRepository) {
             if (StringUtils.isEmpty(jpaUser.getPassword())) {
-                jpaUser.setPassword(getPassword(jpaUser.getUserName()));
+                final String password = getPassword(jpaUser.getUserName());
+                jpaUser.setPassword((password == null) ? PWD_NOLOGON : password);
```

Comment:
Should it not check for password == PWD_NOLOGON here?

Reply:
The current code uses `null` up to the database, where the column is "`NOT NULL`". I kept the usage of null the same, and substitute "nologon" just before storage. This ensures the database constraint stays valid, and rows without a password are visually recognizable.

Reply:
As discussed: `getPassword(username)` will only return `null` if the user is not found. Will change to use `findUser(username)` to make this more explicit.